dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionFromMethodTests1.cs

```diff
@@ -1276,38 +1225,68 @@ public async Task ItShouldNotDeserializeIfParameterTypeAndArgumentTypeAreSameAsy
         Assert.Same(node, actualArgValue);
     }
 
-    [Fact]
-    public async Task ItCanDeserializeJsonStringAsync()
+    [Theory]
+    [InlineData(typeof(string))]
+    [InlineData(typeof(JsonNode))]
+    [InlineData(typeof(JsonDocument))]
+    [InlineData(typeof(JsonElement))]
+    public async Task ItCanDeserializeFunctionArgumentsAsync(Type paramType)
     {
         // Arrange
-        var jsonString = @"{""id"":28}";
+        object? paramValue = paramType switch
+        {
+            Type t when t == typeof(string) => @"{""id"":28}",
+            Type t when t == typeof(JsonNode) => JsonNode.Parse(@"{""id"":28}"),
+            Type t when t == typeof(JsonDocument) => JsonDocument.Parse(@"{""id"":28}"),
+            Type t when t == typeof(JsonElement) => JsonDocument.Parse(@"{""id"":28}").RootElement,
+            _ => throw new ArgumentOutOfRangeException(nameof(paramType))
+        };
+
         CustomTypeForJsonTests? actualArgValue = null;
 
         var func = KernelFunctionFactory.CreateFromMethod((CustomTypeForJsonTests param) => { actualArgValue = param; });
 
         // Act
-        var res = await func.InvokeAsync(this._kernel, new() { ["param"] = jsonString });
+        var res = await func.InvokeAsync(this._kernel, new() { ["param"] = paramValue });
 
         // Assert
         Assert.NotNull(actualArgValue);
         Assert.Equal(28, actualArgValue.Id);
     }
 
-    [Fact]
-    public async Task ItCanDeserializeThirdPartyJsonPrimitivesAsync()
+    [Theory]
+    [InlineData(typeof(string))]
+    [InlineData(typeof(JsonNode))]
+    [InlineData(typeof(JsonDocument))]
+    [InlineData(typeof(JsonElement))]
+    public async Task ItCanUseKernelSerializerOptionsForArgumentsDeserializationAsync(Type paramType)
     {
         // Arrange
-        var thirdPartyJsonPrimitive = new ThirdPartyJsonPrimitive(@"{""id"":28}");
+        object? paramValue = paramType switch
+        {
+            Type t when t == typeof(string) => @"{""id"":28}",
```

Comment:
Can we use a constant for `"{\"id\":28}"` ?

Reply:
Done